plugins/steve/getters.js

```diff
@@ -26,6 +26,81 @@ export default {
     }
   },
 
+  expandedArraySchema: (state, getters) => (type) => {
+    const unwrappedType = type.match(new RegExp(`^${ COLLECTION_TYPES.array }\\[(.*)\\]$`))[1];
+
+    return {
+      type:            COLLECTION_TYPES.array,
+      subType:         unwrappedType,
+      expandedSubType: getters.expandedSchema(unwrappedType)
+    };
+  },
+
+  expandedMapSchema: (state, getters) => (type) => {
+    const unwrappedType = type.match(new RegExp(`^${ COLLECTION_TYPES.map }\\[(.*)\\]$`))[1];
+
+    return {
+      type:            COLLECTION_TYPES.map,
+      subType:         unwrappedType,
+      expandedSubType: getters.expandedSchema(unwrappedType)
+    };
+  },
+
+  expandedResourceFields: (state, getters) => (schema) => {
+    return Object.keys(schema.resourceFields || {}).reduce((agg, key) => {
+      const field = schema.resourceFields[key];
+
+      return {
+        ...agg,
+        [key]: getters.expandedSchema(field)
+      };
+    }, {});
+  },
+
+  expandedSchema: (state, getters) => (typeInput) => {
+    const type = typeof typeInput === 'string'
+      ? typeInput
+      : typeInput.type || typeInput.id;
+
+    if (Object.values(PRIMITIVE_TYPES).includes(type)) {
+      return { type, isPrimitive: true };
+    }
+
+    const schema = getters.schema(type) || { type: typeInput };
+
+    if (type.startsWith(COLLECTION_TYPES.array)) {
+      return getters.expandedArraySchema(type);
+    }
+
+    if (type.startsWith(COLLECTION_TYPES.map)) {
+      return getters.expandedMapSchema(type);
+    }
+
+    if (!schema.resourceFields) {
+      return schema;
+    }
+
+    const keyField = keyFieldFor(type);
+
+    return {
+      ...schema,
+      type:                   schema[keyField],
+      expandedResourceFields: getters.expandedResourceFields(schema)
+    };
+  },
+
+  schema: (state, getters) => (type) => {
```

Comment:
This doesn't affect anything else so I'll merge it, but would like to not have two separate `schemaFor(type)` and `schema(type)` that sound the same (and almost are).  

I haven't stared at it closely but at a glance this looks identical to the fuzzy `schemaFor(type,true)` true except it doesn't use the map to look for an exact match first.